### PR TITLE
Replace TemplateCache w/genericized cache impl

### PR DESCRIPTION
**Pull request by kevsmith**
Cog.Util.Cache & co. unifies Cog's in-memory caches into a single implementation.

### CODE REVIEW

File: lib/cog/util/cache_impl.ex [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    GenServer.call(cacheref, {:delete, key}, :infinity)
+  end
+
+  def close(cacheref) do
+    :erlang.exit(cacheref, :shutdown)
+  end
+
+  def init([args]) do
+    name = Keyword.get(args, :name)
+    ttl = Keyword.get(args, :ttl)
```
Comment (christophermaier) [inline lib/cog/util/cache_impl.ex]: Should these be `Keyword.fetch!/2` instead?

File: lib/cog/util/cache_impl.ex [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  def store(cacheref, key, value) do
+    GenServer.call(cacheref, {:store, key, value}, :infinity)
+  end
+
+  def delete(cacheref, key) do
+    GenServer.call(cacheref, {:delete, key}, :infinity)
+  end
+
+  def close(cacheref) do
+    :erlang.exit(cacheref, :shutdown)
```
Comment (christophermaier) [inline lib/cog/util/cache_impl.ex]: Why not `GenServer.stop/3`?

File: lib/cog/util/cache_impl.ex (hunk 1 above)
Comment (kevsmith) [inline lib/cog/util/cache_impl.ex]: Ah, yeah. Will fix.

File: lib/cog/util/cache_impl.ex (hunk 2 above)
Comment (kevsmith) [inline lib/cog/util/cache_impl.ex]: Because I'm an old fart who forgets such niceties exist? ;-)

**Pull request merged**